onnxruntime/hosting/http/session.cc

```diff
@@ -93,31 +93,34 @@ void HttpSession::Send(Msg&& msg) {
 }
 
 template <typename Body, typename Allocator>
-void HttpSession::HandleRequest(boost::beast::http::request<Body, boost::beast::http::basic_fields<Allocator> >&& req) {
+void HttpSession::HandleRequest(http::request<Body, http::basic_fields<Allocator> >&& req) {
   HttpContext context{};
-  context.request = req;
+  context.request = std::move(req);
 
-  std::string path = req.target().to_string();
-  std::string model_name;
-  std::string model_version;
-  std::string action;
+  std::string path = context.request.target().to_string();
+  std::string model_name, model_version, action;
   handler_fn func;
-  http::status status = routes_->ParseUrl(req.method(), path, model_name, model_version, action, func);
+  http::status status = routes_->ParseUrl(context.request.method(), path, model_name, model_version, action, func);
+
+  // TODO: set request id
```

Comment:
I think it is too late to set request id. Assign request id should be the action right after we receive this request. Then any other logic in our code could have log with the unique request id. It will be very useful for issue investigation.

Reply:
This is essentially where we receive the request. We can move it above `ParseUrl` but not higher.

Reply:
Please have it above the `ParseUrl` to make sure we could log error if it is url related.

Reply:
Moved the comment to use it above ParseUrl.